tests/test_defects.py

```diff
@@ -0,0 +1,270 @@
+        for defect in defects:
+            self.assertNotIn(defect.getBBox(), shouldBeMissed)
+
+    def test_postProcessDefectSets(self):
+        defects = self.defaultTask.findHotAndColdPixels(self.flatExp, 'flat')
+
+        # minimal test
+        merged = self.defaultTask._postProcessDefectSets([defects], self.flatExp.getDimensions(), 'FRACTION')
+        self.assertEqual(defects, merged)
+
+        # should always be true regardless of config
+        merged = self.defaultTask._postProcessDefectSets([defects, defects], self.flatExp.getDimensions(),
+                                                         'FRACTION')
+        self.assertEqual(defects, merged)
+
+        # now start manipulating defect lists
+        config = copy.copy(self.defaultConfig)
+        config.combinationMode = 'FRACTION'
+        config.combinationFraction = 0.85
+        task = cpPipe.defects.FindDefectsTask(config=config)
+        merged = task._postProcessDefectSets([defects, defects], self.flatExp.getDimensions(), 'FRACTION')
+
+        defectList = [defects]*10  # 10 identical defect sets
+        # remove one defect from one of them, should still be over threshold
+        defectList[7] = defectList[7][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'FRACTION')
+        self.assertEqual(defects, merged)
+
+        # remove another and should be under threshold
+        defectList[3] = defectList[3][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'FRACTION')
+        self.assertNotEqual(defects, merged)
+
+        # now test the AND and OR modes
+        defectList = [defects]*10  # 10 identical defect sets
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertEqual(defects, merged)
+
+        defectList[7] = defectList[7][:-1]
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'AND')
+        self.assertNotEqual(defects, merged)
+
+        merged = task._postProcessDefectSets(defectList, self.flatExp.getDimensions(), 'OR')
+        self.assertEqual(defects, merged)
+
+    def test_pixelCounting(self):
+        config = copy.copy(self.defaultConfig)
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+        task = cpPipe.defects.FindDefectsTask(config=config)
+        defects = task.findHotAndColdPixels(self.flatExp, 'flat')
+
+        defectArea = 0
+        for defect in defects:
+            defectArea += defect.getBBox().getArea()
+
+        crossCheck = 0
+        for x, y, s in self.brightDefects:
+            crossCheck += s**2
+        for x, y, s in self.darkDefects:
+            crossCheck += s**2
+
+        self.assertEqual(defectArea, task._nPixFromDefects(defects))  # the real function being tested
+        self.assertEqual(defectArea, crossCheck)  # but calculate the sum two ways
```

Comment:
I don't understand what advantage these two asserts give? If `getBBox().getArea()` is wrong, we have bigger problems...

Reply:
Summing the areas is not the same as looking at the number of defective pixels in an image, for several reasons: overlapping bboxes, pixels not making the cut when defect sets are merged, maybe others. Definitely doesn't hurt to assert twice - if it's always true then no harm, and if it's not then it should be looked at.

Reply:
I think the "the real function being tested" comment threw me off some. Maybe remove those two inline comments and replace them with a comment over the whole thing:

"Test the result of `_nPixFromDefects` via two different ways of calculating area."

Reply:
Done.